packages/@cdktf/provider-generator/lib/get/constructs-maker.ts

```diff
@@ -6,21 +6,23 @@ import { CodeMaker } from "codemaker";
 import { mkdtemp } from "../util";
 import * as srcmak from "jsii-srcmak";
 import {
-  TerraformModuleConstraint,
   TerraformDependencyConstraint,
   logger,
   logTimespan,
+  ConstructsMakerProviderTarget,
+  ConstructsMakerModuleTarget,
+  ConstructsMakerTarget,
+  ProviderSchema,
+  ModuleSchema,
+  Errors,
 } from "@cdktf/commons";
 import { DISPLAY_VERSION, Language } from "@cdktf/commons";
-import {
-  ProviderSchema,
-  readProviderSchema,
-  readModuleSchema,
-} from "./generator/provider-schema";
 import { TerraformProviderGenerator } from "./generator/provider-generator";
 import { ModuleGenerator } from "./generator/module-generator";
-import { ModuleSchema } from "./generator/module-schema";
 import { glob } from "glob";
+import { readSchema } from "@cdktf/provider-schema";
+
+type Awaited<T> = T extends PromiseLike<infer U> ? U : T;
```

Comment:
nit: I'm sure you found the same StackOverflow post, but doesn't this shadow the built-in `Awaited`?
https://stackoverflow.com/a/49889856

Reply:
This seems to be copied old code that probably predated us using TS 4.5+ 😂